cmd/scollector/collectors/elasticsearch.go

```diff
@@ -35,189 +41,228 @@ func AddElasticIndexFilter(s string) error {
 	if err != nil {
 		return err
 	}
-	esIndexFilters = append(esIndexFilters, re)
+	elasticIndexFilters = append(elasticIndexFilters, re)
 	return nil
 }
 
+type structProcessor struct {
+	elasticVersion string
```

Comment:
In this case, yes. The elastic version processing is what first pushed me down this road.

Would definitely like to consider a generalized version for certain other collectors where it makes sense. If I build that I can maybe have the elastic collector use that while tacking on the version logic.